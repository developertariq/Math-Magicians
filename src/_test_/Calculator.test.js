// import React from 'react';
// import renderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
// import { BrowserRouter as Router } from 'react-router-dom';
// import '@testing-library/jest-dom';
// import Calculator from '../components/Calculator';
// import Calculation from '../pages/Calculation';

// describe('Calculator page', () => {
//   test('renders Calculator page correctly', () => {
//     const tree = renderer.create(<Calculation />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   test('renders Calculator correctly', () => {
//     const tree = renderer.create(<Calculator />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   test('renders the correct Calculator', () => {
//     const { getByTestId } = render(
//       <Router>
//         <Calculator />
//       </Router>,
//     );
//     const calculator = getByTestId('calculator');
//     expect(calculator).toBeInTheDocument();
//   });
// });
