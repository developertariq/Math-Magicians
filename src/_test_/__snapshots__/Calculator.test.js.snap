// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page renders Calculator correctly 1`] = `
<div
  className="calculator"
  data-testid="calculator"
>
  <div
    className="display"
  />
  <div
    className="keypad"
  >
    <button
      className="key"
      data-value="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key"
      data-value="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key"
      data-value="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key color-button"
      data-value="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="key"
      data-value="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key"
      data-value="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key"
      data-value="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key color-button"
      data-value="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="key"
      data-value="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key"
      data-value="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key"
      data-value="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key color-button"
      data-value="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="key"
      data-value="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key"
      data-value="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key"
      data-value="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key color-button"
      data-value="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="key double-size"
      data-value="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key"
      data-value="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key color-button"
      data-value="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Calculator page renders Calculator page correctly 1`] = `
<div
  className="main-container"
  style={
    Object {
      "display": "flex",
      "justifyContent": "space-between",
    }
  }
>
  <h2
    className="page-title"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
    data-testid="calculator"
  >
    <div
      className="display"
    />
    <div
      className="keypad"
    >
      <button
        className="key"
        data-value="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="key"
        data-value="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="key"
        data-value="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="key color-button"
        data-value="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="key"
        data-value="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="key"
        data-value="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="key"
        data-value="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="key color-button"
        data-value="x"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="key"
        data-value="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="key"
        data-value="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="key"
        data-value="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="key color-button"
        data-value="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="key"
        data-value="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="key"
        data-value="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="key"
        data-value="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="key color-button"
        data-value="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="key double-size"
        data-value="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="key"
        data-value="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="key color-button"
        data-value="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
